airflow-core/src/airflow/api_fastapi/core_api/datamodels/dag_run.py

```diff
@@ -163,3 +163,10 @@ class DAGRunsBatchBody(StrictBaseModel):
     start_date_lte: AwareDatetime | None = None
     end_date_gte: AwareDatetime | None = None
     end_date_lte: AwareDatetime | None = None
+
+
+class DAGRunWatchResult(StrictBaseModel):
```

Comment:
Do we really need a new Serializer? I'm not fan of having each endpoint returning custom schema, unless that would really impact / slow down performances. Here can't we just use the default `DAGRunResponse`, which holds the state and duration. This will be much easier for client to understand the API if we avoid multiplication of partial schemas.